src/mlpack/core/optimizers/adam/adam_impl.hpp

```diff
@@ -113,19 +119,44 @@ double Adam<DecomposableFunctionType>::Optimize(arma::mat& iterate)
     m *= beta1;
     m += (1 - beta1) * gradient;
 
-    v *= beta2;
-    v += (1 - beta2) * (gradient % gradient);
+    if (adaMax)
+    {
+      u *= beta2;
+      arma::mat absGradient = arma::abs(gradient);
+      // Update the exponentially weighted infinity norm.
+      for (size_t row = 0; row != iterate.n_rows; ++row)
+      {
+        for (size_t col = 0; col != iterate.n_cols; ++col)
+        {
+          if (u.at(row, col) < absGradient.at(row, col))
+            u.at(row, col) = absGradient.at(row, col);
+        }
+      }
+    }
+    else
+    {
+      v *= beta2;
+      v += (1 - beta2) * (gradient % gradient);
```

Comment:
It is not related to AdaMax but since it concerns the same code so asking here itself -- according to the paper, v should calculated as `v += (1 - beta2) * arma::pow(gradient, 2)` but here we take the modulo instead of multiplying it with itself. I can't tell but is there an advantage of doing that in some way?